### PR TITLE
[CP]: Lowest Common Ancestor in Binary Tree

### PR DESCRIPTION
**Pull request by areenoverclouds**
<hr>

## Description 📜

This is the documentation about lowest common ancestor problem.
Fixes ##7966

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [x] Competitive Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: Competitive_Programming/C++/Binary Trees/lca.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    2. If root == node1 or root == node2, then return root.
+    3. Recursively call the function for left and right subtrees.
+    4. If both the nodes are in left or right subtree exclusively then we will return the left or right subtree respectively.
+    5. If current node is NULL then we have reached the end of tree.
+
+## Code
+
+The following code snippet shows the LCA function used to find lowest commmon ancestor.
+
+**Method 1:**
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Binary Trees/lca.md:48]: name this approach here as well.

File: Competitive_Programming/C++/Binary Trees/lca.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    4. Traverse the vectors till elements are same.
+    5. Return the last same element.
+
+* Second method: Recursive Approach. This method can be adopted to find LCA in a single traversal. The time complexity remains O(n).
+    ### Algorithm
+    1. Traverse tree from root.
+    2. If root == node1 or root == node2, then return root.
+    3. Recursively call the function for left and right subtrees.
+    4. If both the nodes are in left or right subtree exclusively then we will return the left or right subtree respectively.
+    5. If current node is NULL then we have reached the end of tree.
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Binary Trees/lca.md:42]: Place this _below_ approach-1's code and _above_ approach-2's code.
Like:

Description
Code
Description
Code

File: Competitive_Programming/C++/Binary Trees/lca.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+     else if(left_subtree!=NULL)
+        return left_subtree;
+     else
+        return right_subtree;
+
+     return NULL;
+}
+```
+
+## Complexity Analysis
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Binary Trees/lca.md:108]: Add the Complexity details for both the approaches

**Review by ravireddy07 — CHANGES_REQUESTED** [review]

**Comment by areenoverclouds** [comment]
@ravireddy07 I have made the changes, please recheck.

**Review by ravireddy07 — APPROVED** [review]

**Comment by ravireddy07** [comment]
Please update the status in PR Tracksheet @areenoverclouds

**Pull request merged**